### PR TITLE
Allow JSON deserialization into JsonWebKeySet

### LINKED ISSUE #2326: Unable to deserialize JSON into JsonWebKeySet
Currently, it is not possible to deserialize a JWK Set into `Microsoft.IdentityModel.Tokens.JsonWebKeySet` without additionnal configuration.

This is because the `Keys` property is readonly.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/eeaa61ba4887ded80875ed5659aa8a7c04b1da96/src/Microsoft.IdentityModel.Tokens/JsonWebKeySet.cs#L82

Steps to reproduce:
```cs
using System.Net.Http.Json;
using Microsoft.IdentityModel.Tokens;

var client = new HttpClient();
var response = await client.GetAsync("https://login.microsoftonline.com/common/discovery/v2.0/keys");
var jsonWebKeySet = await response.Content.ReadFromJsonAsync<JsonWebKeySet>();
var count = jsonWebKeySet!.Keys.Count; // 0
```

**Known workarounds**

Before .NET 8, we can use `JsonWebKeySet.Create(string)` to create an instance of `JsonWebKeySet` from the raw content:

```cs
using System.Net.Http.Json;
using Microsoft.IdentityModel.Tokens;

var client = new HttpClient();
var response = await client.GetAsync("https://login.microsoftonline.com/common/discovery/v2.0/keys");
var raw = await response.Content.ReadAsStringAsync(); // An temporary string is allocated here :(
var jsonWebKeySet = JsonWebKeySet.Create(raw);
var count = jsonWebKeySet!.Keys.Count; // 5
```

With .NET 8, we can use the new [`JsonObjectCreationHandling`](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonobjectcreationhandling) introduced, this way:

```cs
using System.Net.Http.Json;
using System.Text.Json.Serialization;
using System.Text.Json;
using Microsoft.IdentityModel.Tokens;

var client = new HttpClient();
var response = await client.GetAsync("https://login.microsoftonline.com/common/discovery/v2.0/keys");
var jsonWebKeySet = await response.Content.ReadFromJsonAsync<JsonWebKeySet>(new JsonSerializerOptions
{
    PreferredObjectCreationHandling = JsonObjectCreationHandling.Populate
});
var count = jsonWebKeySet!.Keys.Count; // 5
```

**Suggested solution**
 
We can add the attribute `JsonObjectCreationHandlingAttribute` with the value `JsonObjectCreationHandling.Populate` directly on the `Keys` property to allow the deserialization without additional configuration.
This attribute has been introduced in .NET 8 so it will probably be necessary to condition it with `if NET8_0_OR_GREATER`.

### PR DESCRIPTION
**Pull request by ghost**
Fixes #2326.

We'll probably have to wait for .NET 8 before we can merge this PR.

### CODE REVIEW

File: test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeySetTests.cs [hunk 1]
```diff
@@ -417,6 +417,15 @@ public static TheoryData<JsonWebKeySetTheoryData> GetSigningKeysTheoryData
             }
         }
 
+        [Fact]
+        public void JsonDeserialize()
```
Comment (joegoldman2) [inline test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeySetTests.cs:421]: I'm not really sure what can be the right strategy to run this test only if the TFM is `.net8.0`. I'm open to suggestions :)

Comment (joegoldman2) [inline test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeySetTests.cs:421]: I'm not really sure what can be the right strategy to run this test only if the TFM is `net8.0`. I'm open to suggestions :)

Comment (ghost) [inline test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeySetTests.cs:422]: I'm not really sure what can be the right strategy to run this test only if the TFM is `net8.0` and above, apart from putting a `#if NET8_0_OR_GREATER` around the test. I'm open to suggestions :)

**Comment by joegoldman2** [comment]
@jennyf19 @brentschmaltz this PR is ready to be reviewed.

File: test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeySetTests.cs (hunk 1 above)
Comment (brentschmaltz) [reply test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeySetTests.cs:422]: @joegoldman2 yes you can put NET_8_OR_GREATER

Comment (joegoldman2) [reply test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeySetTests.cs:422]: Perfect, done!

**Review by brentschmaltz — APPROVED** [review]

**Pull request merged**